.circleci/config.yml

```diff
@@ -65,102 +87,395 @@ jobs:
 
 workflows:
   version: 2
-  # This one is run only when triggered with the API, using a GitHub action
-  pull_request:
-    when: << pipeline.parameters.run_pull_request >>
+  # On code change is executed each time new code is pushed to a branch
+  # Current project configuration in circleci only builds on PR + main, so effectively this is only being executed
+  on-code-change:
     jobs:
-      - checkout:
-          name: PR Checkout
+      - initialize
       - build:
-          name: PR Build
-          is_pull_request: true
           context: QA_ENVIRONMENT
           requires:
-            - PR Checkout
+            - initialize
       - sonar:
-          name: PR Sonar analysis
-          is_pull_request: true
           context: QA_ENVIRONMENT
           requires:
-            - PR Build
-  on-code-change:
-    jobs:
-      - checkout:
+            - build
+      - publish:
+          context: QA_ENVIRONMENT
+          requires:
+            - build
           filters:
             branches:
               only:
-                - master
-                - /feature-.*/
-                - /[0-9]_[0-9]_x/
-      - build:
-          is_pull_request: false
+                - << pipeline.parameters.PRIMARY_RELEASE_BRANCH >>                
+      # Verifies that the new code works with the latest released version of Jahia
+      - integration_tests:
+          name: 'Int. Tests - Jahia Latest - Built modules'
+          requires:
+            - build
+          context: QA_ENVIRONMENT
+          #Using Aliases here (8 is the latest of the 8 tree, 8.1 is the latest of the tree, ...)
+          JAHIA_IMAGE: jahia/jahia-dev:8.0.1.0
+          RUN_TESTSIMAGE: << pipeline.parameters.TESTS_IMAGE >>:latest
+          TOOLS_USER: root
+          TOOLS_PWD: root
+          MANIFEST: warmup-manifest-release.yml
+          TESTRAIL_MILESTONE: Jahia-Latest
+          SKIP_TESTRAIL: "true"
+          SKIP_ARTIFACTS: false
+          BUILD_TESTSIMAGE: true
+      - docker-publish:
           context: QA_ENVIRONMENT
           requires:
-            - checkout
-      - sonar:
-          name: Sonar analysis
-          is_pull_request: false
+            - Int. Tests - Jahia Latest - Built modules
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              only:
+                - << pipeline.parameters.PRIMARY_RELEASE_BRANCH >>
+
+  # The objective of nightly runs is to ensure the codebase still passes tests:
+  # - With the latest releases version of Jahia
+  # - With the current snapshot of Jahia
+  nightly:
+    triggers:
+      - schedule:
+          cron: '0 0 * * *'
+          filters:
+            branches:
+              only:
+                - << pipeline.parameters.PRIMARY_RELEASE_BRANCH >>
+    jobs:
+      - initialize
+      - integration_tests:
+          matrix:
+            parameters:
+              JAHIA_IMAGE: ["jahia/jahia-private:snapshot", "jahia/jahia-dev:8.0"]
```

Comment:
Just a quick note to rename `jahia/jahia-private:snapshot` to `jahia/jahia-private:snapshot-dev`